metals/src/main/scala/scala/meta/internal/bsp/ConnectionBspStatus.scala

```diff
@@ -55,4 +116,76 @@ object ConnectionBspStatus {
       command = ServerCommands.ConnectBuildServer.id,
       commandTooltip = "Reconnect.",
     ).withStatusType(StatusType.bsp)
+
+  def bspErrorParams(
+      serverName: String,
+      icons: Icons,
+      message: String,
+      errorsNumber: Int,
+  ): MetalsStatusParams =
+    MetalsStatusParams(
+      s"$serverName $errorsNumber ${icons.alert}",
+      "warn",
+      show = true,
+      tooltip = message.trimTo(TOOLTIP_MAX_LENGTH),
+      command = ServerCommands.RunDoctor.id,
+      commandTooltip = "Open doctor.",
+    ).withStatusType(StatusType.bsp)
+
+  private val TOOLTIP_MAX_LENGTH = 150
+}
+
+trait BspServerState
+case object Disconnected extends BspServerState
+case class Connected(serverName: String) extends BspServerState
+case class ErrorMessage(message: String) extends BspServerState
+case object NoResponse extends BspServerState
+
+case class BspStatusState(
+    currentState: BspServerState,
+    previousState: Option[BspServerState],
```

Comment:
```suggestion
    lastError: Option[ErrorMessage],
```
I think we only use it as an error?